pyiron/atomistics/job/structurecontainer.py

```diff
@@ -2,62 +2,160 @@
+                "AtomisticGenericJob object."
             )
-        if self.status.initialized:
-            self.run()
 
-    def append(self, structure_to_append):
-        self.structure = structure_to_append
+    @property
+    def structure(self):
+        """
+        :class:`~.Atoms`: first (or only) structure set in the container
+        """
+        return self.structures[0].atoms
+
+    @structure.setter
+    def structure(self, structure):
+        item = self.__to_structure(structure_or_job)
+        if len(self.structures) >= 1:
+            self.structures[0] = item
+        else:
+            self.structures.append(item)
+
+    def append(self, structure_or_job, **kwargs):
+        """
+        Add new structure to structure list.
+
+        The added structure will available in
+        :attr:`~.StructureList.structures`.  If **kwargs contains a key "atoms"
+        is ignored, since the structure is saved under this key.  If the
+        structure is added via a job, retrieve the latest structure and its id
+        is also saved under the "jobid" key and **kwargs of the same name are
+        ignored.
+
+        Args:
+            structure_or_job (AtomisticGenericJob/Atoms):
+                if :class:`~.AtomisticGenericJob` add from
+                :meth:`~.AtomisticGenericJob.get_structure`,
+                otherwise add just the given :class:`~.Atoms`
+            **kwargs: meta data tags to add to the structure in the list
+
+        Returns:
+            dict: item added to :attr:`~.structures`
+        """
+
+        kwargs.update(self.__to_structure(structure_or_job))
+        self.__structures.append(kwargs)
+
+        return self.structures[0]
+
+    def run_static(self):
+        self.status.finished = True
 
     def run_if_interactive(self):
+        self.to_hdf()
         self.status.finished = True
 
-    def to_hdf(self, hdf=None, group_name=None):
-        super(StructureContainer, self).to_hdf(hdf=hdf, group_name=group_name)
-        with self.project_hdf5.open("input") as hdf5_input:
-            self.structure.to_hdf(hdf5_input)
+    def write_input(self):
+        pass
+
+    def collect_output(self):
+        pass
+
+    def to_hdf(self, hdf = None, group_name = None):
+        # skip any of the AtomisticGenericJob specific serialization, since we
+        # handle the structures on our own and that method might just confuse
+        # self.structure and self.structures
+        super(GenericJob, self).to_hdf(hdf = hdf, group_name = group_name)
+
+        hdf = self.project_hdf5.create_group("structures")
+
+        for i, s in enumerate(self.__structures.values()):
+            sub = hdf.create_group("index_{}".format(i))
+            for k, v in s.items():
+                if k == "atoms":
+                    # use to_hdf from Atoms class
+                    v.to_hdf(hdf = sub, group_name = "atoms")
+                else:
+                    # otherwise use the normal hdfio implementation
```

Comment:
This is dangerous, because by default it creates a separate HDF5 node for each key, as our nodes have a rather large byte length we are wasting a lot of space. But you can also store the whole dictionary in HDF5 - once he structure is removed - then h5io is going to store it as a single JSON string. This is what we do for the server object 
https://github.com/pyiron/pyiron/blob/master/pyiron/base/server/generic.py#L410

Reply:
My intention was to make filtering the structures before loading the whole jobs as easy as possible.

As a compromise I just tried setting the attributes of the structure group instead of creating new datasets for each tag, that reduces the final size by ~3.  Naturally the json is still ~10 smaller than this.

Reply:
I like this idea - still loading the full dictionary is also available in inspect mode, so that should not be an issue to filter the structures.